transmart-notifications/grails-app/jobs/org/transmart/notifications/QuerySetSubscriptionDailyJob.groovy

```
@@ -13,21 +14,47 @@ import org.transmartproject.core.userquery.SubscriptionFrequency
 class QuerySetSubscriptionDailyJob {
 
     QuerySetSubscriptionMailService querySetSubscriptionMailService
+
+    /**
+     * Parse org.transmart.notifications.dailyJobTriggerTime setting specified in the configuration
+     * Format of the setting is hh-mm, which represent hour and minutes, separated by '-'.
+     * Hours and minutes have to be numbers from a proper range (0-23 and 0-59 respectively).
+     * Specified setting of hour and minutes is changed to a proper cron expression that will be triggered daily.
+     * If the specified setting is invalid, the default one will be set, which is 00:00 every day.
+     *
+     * @return valid cron expression for a daily job
+     */
+    private static String parseConfiguredTriggerTime() {
+        try {
+            def (hour, minute) = Holders.config.org.transmart.notifications.dailyJobTriggerTime.tokenize('-')
+            def validMinute = minute as Integer
+            def validHour = hour as Integer
+            if (validMinute >= 0 && validMinute <= 59 && validHour >= 0 && validHour <= 23){
+                return " 0 $minute $hour 1/1 * ?"
+            } else {
+                throw new InvalidArgumentsException('Property out of allowed range.')
+            }
+        } catch (Exception e) {
+            log.warn "Invalid trigger time setting in transmart-notification configuration.", e.message
```

Just fail instead of writing a warning. The config should parse correctly.